### PR TITLE
[PARKED][AIRFLOW-2912] [WIP] Add invoke operator for Google Cloud Functions

### PR DESCRIPTION
**Pull request by sprzedwojski**
I've revisited the invoke operator that was done a few months ago, rebased it on top of the current master branch and applied some necessary changes and refactorings.

### CODE REVIEW

File: airflow/contrib/hooks/gcp_function_hook.py [hunk 1]
```diff
@@ -193,3 +205,109 @@ def _wait_for_operation_to_complete(self, operation_name):
...
+class IAPRequest(object):
+    """
+    This class is based on the example in the IAP docs and modified to match our needs.
+    Source: https://cloud.google.com/iap/docs/authentication-howto#iap_make_request-python
+    """
+    def make_iap_request(self, url,
+                         credentials,
+                         method='GET', **kwargs):
+        """
+        Makes a request to an application protected by Identity-Aware Proxy.
```
Comment (Xe0n0) [inline airflow/contrib/hooks/gcp_function_hook.py]: nits: Makes a request to a url protected by...

File: airflow/contrib/operators/gcp_function_operator.py [hunk 2]
```diff
@@ -315,3 +316,73 @@ def execute(self, context):
             else:
                 self.log.error('An error occurred. Exiting.')
                 raise e
+
+
+class GcfFunctionInvokeOperator(BaseOperator):
+    """
+    Invokes an HTTP-triggered, IAM-protected Cloud Function under a given URL.
```
Comment (Xe0n0) [inline airflow/contrib/operators/gcp_function_operator.py]: maybe nits: Invokes a Cloud Function via by IAM-protected HTTP trigger URL.

File: airflow/contrib/example_dags/example_gcp_function.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+# specific language governing permissions and limitations
+# under the License.
+
+"""
+Example Airflow DAG that displays interactions with Google Cloud Functions.
+It creates a function, invokes and then deletes it.
+
+This DAG relies on the following OS environment variables
+https://airflow.apache.org/concepts.html#variables
+* PROJECT_ID - Google Cloud Project to use for the Cloud Function.
```
[the remark this answers is not in the document]
Comment (sprzedwojski) [reply airflow/contrib/example_dags/example_gcp_function.py]: Done

File: airflow/contrib/hooks/gcp_function_hook.py [hunk 4]
```diff
@@ -168,6 +165,21 @@ def delete_function(self, name):
         operation_name = response["name"]
         return self._wait_for_operation_to_complete(operation_name)
 
+    def invoke_function(self, url):
+        """
+        Invokes an HTTP-triggered, IAM-protected Cloud Function under a given URL.
+        Credentials used to authorize to the IAM Function are the credentials provided
+        for the GCP connection.
+
+        :param url: The Identity-Aware Proxy-protected URL to fetch.
```
[the remark this answers is not in the document]
Comment (sprzedwojski) [reply airflow/contrib/hooks/gcp_function_hook.py]: Done

File: airflow/contrib/operators/gcp_function_operator.py (hunk 2 above)
Comment (sprzedwojski) [reply airflow/contrib/operators/gcp_function_operator.py]: Done

**Comment by nuclearpinguin** [comment]
Closed via https://github.com/apache/airflow/pull/5995

**Pull request closed without merging**